### ISSUE TITLE
Orchestrator should redeem ticket even when face value exceeds broadcaster max float under certain conditions

### ISSUE THREAD

**Issue by yondonfu** [livepeer/go-livepeer#1581]
**Is your feature request related to a problem? Please describe.**

An orchestrator currently only redeems a winning ticket if:

- The broadcaster's [max float is not 0](https://github.com/livepeer/go-livepeer/blob/master/pm/sendermonitor.go#L300)
- The broadcaster's [max float is insufficient to cover the ticket face value](https://github.com/livepeer/go-livepeer/blob/0723bac6873f09e08c71ad7bf5f98f0466db97fc/pm/sendermonitor.go#L309)

This can result in a situation where an orchestrator will let a winning ticket expire because the broadcaster's max float decreases (i.e. due to a increase in the # of active orchestrators) such that it is no longer sufficient to cover the ticket's face value. This could happen even if the broadcaster has enough funds in its deposit to cover the ticket face value. 

**Describe the solution you'd like**

An orchestrator should redeem a winning ticket as long as the broadcaster's max float + the broadcaster's deposit is >= the ticket face value.

Note that there is a risk with this condition in that if the broadcaster's deposit is drained before the orchestrator's redemption tx confirms then the orchestrator may actually lose value with the redemption tx if the tx cost is not covered. The second condition described below would address this, but likely requires some code shuffling because the `LocalSenderMonitor` responsible for redeeming winning tickets does not currently have access to the estimated tx cost of a redemption (only the `Recipient` has access to this right now).

Other conditions worth considering:

- The ticket face value > the estimated tx cost because the estimated tx cost could be different at the time of redemption vs. the time at which the ticket face value was originally calculated
- The broadcaster's max float >= the estimated tx cost because this would mean that orchestrator is at least guaranteed that the the estimated tx cost will be covered by the redemption tx

**Comment by yondonfu** [comment]
Below is a spec for a proposed solution based on some thinking from #943:

# Redeem Based on Available Funds and Tx Cost

## Abstract


An orchestrator can redeem tickets as long the available funds for a broadcaster exceeds the estimated transaction cost for redeeming a ticket.

## Motivation


See the OP in this thread and #943.

TLDR: The conditions for redeeming a ticket can cause an orchestrator to allow tickets to expire (no longer redeemable) even if the orchestrator could gain value from redeeming the tickets (relative to the transaction cost of redemption).

## Proposed Solution


When an orchestrator prepares to redeem a ticket in `LocalSenderMonitor.redeemWinningTicket()`:

- Let `availFunds = broadcaster.reserveAlloc + broadcaster.deposit - broadcaster.pendingAmount`
- Let `txCost` be the current estimated transaction cost for redeeming a ticket
- If `availFunds > txCost`, redeem the ticket
- Else, try the redemption later

The rationale for using `availFunds` instead of the broadcaster's max float for the check is that at this point the orchestrator already has a winning ticket from the broadcaster. So, the orchestrator should base the decision of whether or not to redeem a ticket based on the total funds available that could be used to compensate it. This is in contrast with a broadcaster's max float which is used to determine the max "safe" face value for an orchestrator to use. If an orchestrator already has a winning ticket, whether or not the face value is "safe" matters less than whether the available funds of a broadcaster could be used to compensate the orchestrator.

The rationale for the check being `availFunds  > txCost` instead of `availFunds >= txCost` is that in practice, there are other costs incurred by the orchestrator in order to ultimately withdraw value received from a broadcaster (i.e. the transaction cost for claiming earnings and withdrawing). I opted not to try to account for these other costs for the sake of simplicity for now, but figured that it would at least be reasonable to say that "it is not enough for txCost to be covered, the orchestrator needs to receive *more* than the txCost".

## Implementation Tasks and Considerations


- Support querying the current estimated gas price and accessing the estimated redeem gas in `LocalSenderMonitor`
    - There are already a lot of parameters used to initialize `LocalSenderMonitor`, so a config struct might be helpful here
- Update the [checks](https://github.com/livepeer/go-livepeer/blob/82509ae41b822a3ab54e266315eb6a9dd0a692ff/pm/sendermonitor.go#L300) in `LocalSenderMonitor.redeemWinningTicket()` to use the logic described above

## Testing Tasks and Considerations


- Update unit tests
- Make sure that the scenario described in the OP is no longer an issue
